nav2_docking/opennav_docking/src/simple_charging_dock.cpp

```diff
@@ -322,6 +350,88 @@ void SimpleChargingDock::jointStateCallback(const sensor_msgs::msg::JointState::
   is_stalled_ = (velocity < stall_velocity_threshold_) && (effort > stall_effort_threshold_);
 }
 
+void SimpleChargingDock::startDetection()
+{
+  // Skip if already starting or ON
+  if (detector_state_ != DetectorState::OFF) {return;}
+
+  // 1. Service START request
+  if (detector_client_) {
+    auto req = std::make_shared<std_srvs::srv::Trigger::Request>();
+    try {
+      detector_client_->invoke(
+        req,
+        std::chrono::duration_cast<std::chrono::nanoseconds>(
+          std::chrono::duration<double>(detector_service_timeout_)));
+    } catch (const std::exception & e) {
+      RCLCPP_WARN(node_->get_logger(),
+        "Detector service '%s' failed: %s",
+        detector_service_name_.c_str(), e.what());
+    }
+  }
+
+  // 2. Subscription toggle
+  //    Only subscribe once; will set state to ON on first message
+  if (subscribe_toggle_ && !detected_pose_sub_) {
+    detected_pose_sub_ = node_->create_subscription<geometry_msgs::msg::PoseStamped>(
+      "detected_dock_pose", rclcpp::SensorDataQoS(),
+      [this](const geometry_msgs::msg::PoseStamped::SharedPtr pose) {
+        detected_dock_pose_ = *pose;
+        detector_state_ = DetectorState::ON;
+      });
+  }
+
+  detector_state_ = DetectorState::ON;
+  RCLCPP_INFO(node_->get_logger(), "Detector START requested, state set to ON");
+}
+
+void SimpleChargingDock::stopDetection()
+{
+  // Skip if already OFF
+  if (detector_state_ == DetectorState::OFF) {return;}
+
+  // 1. Service STOP request
+  if (detector_client_) {
+    auto req = std::make_shared<std_srvs::srv::Trigger::Request>();
+    try {
+      detector_client_->invoke(
+        req,
+        std::chrono::duration_cast<std::chrono::nanoseconds>(
+          std::chrono::duration<double>(detector_service_timeout_)));
+    } catch (const std::exception & e) {
+      RCLCPP_WARN(node_->get_logger(),
+        "Detector service '%s' failed: %s",
+        detector_service_name_.c_str(), e.what());
+    }
+  }
+
+  // 2. Unsubscribe to release resources
+  //    reset() will tear down the topic subscription immediately
+  if (subscribe_toggle_ && detected_pose_sub_) {
+    detected_pose_sub_.reset();
+  }
+
+  detector_state_ = DetectorState::OFF;
+  RCLCPP_INFO(node_->get_logger(), "Detector STOP requested");
+}
+
+void SimpleChargingDock::activate()
+{
+  RCLCPP_DEBUG(node_->get_logger(), "SimpleChargingDock activated");
+}
+
+void SimpleChargingDock::deactivate()
+{
+  RCLCPP_DEBUG(node_->get_logger(), "SimpleChargingDock deactivated");
```

Comment:
If you make deactivated call `stopDetection`, you can remove the explicit `stopDetection` call in the dock database. Its probably good that these reset its own resources internally.

Reply:
@bkoensgen